src/main/java/com/sugarcrm/candybean/automation/webdriver/WaitConditions.java

```diff
@@ -337,4 +410,81 @@ public String toString() {
 	public static ExpectedCondition<Boolean> textIsPresentInValue(WebDriverElement wde, String text) {
 		return ExpectedConditions.textToBePresentInElementValue(wde.we, text);
 	}
+
+	/**
+	 * Return the element found by hook that contains the specified attribute value if expected,
+	 * or the reverse if not.
+	 *
+	 * @param hook	The hook used to find the element
+	 * @param attribute	The attribute to check
+	 * @param value	The expected value of the attribute
+	 * @param expectValue If the value is expected or not
+	 * @return The element if the specified value contains the specified attributed, null otherwise
+	 */
+	public static ExpectedCondition<WebDriverElement> hasAttribute(final Hook hook, final String attribute,
+																   final String value, final boolean expectValue) {
+		return new ExpectedCondition<WebDriverElement>() {
+			@Override
+			public WebDriverElement apply(WebDriver driver) {
+				try {
+					WebElement element = findElement(hook, driver);
+					/*
+					Split the string so that we can match the attribute.
+					If we are waiting for a value of "red", then "red left-aligned large"
+					should match that, but "starred" should not.
+					 */
+					String[] values = element.getAttribute(attribute).split("\\s");
+					for (int x = 0; x < values.length; x++) {
```

Comment:
We generally prefer foreach syntax, e.g. "for(String value : values)" for readability.